recipes-core/base-files/base-files_%.bbappend

```diff
@@ -0,0 +1,9 @@
+do_install_append_rpi () {
+    # ensure that files and directories are created as required
+    mkdir -p ${D}${sysconfdir}/modules-load.d/
+    touch ${D}${sysconfdir}/modules-load.d/i2c_dev.conf
```

Comment:
This touch can go away. You only need to make sure the parent directory is created.